appsdk/configurable.go

```diff
@@ -230,20 +231,36 @@ func (dynamic AppFunctionsSDKConfigurable) CompressWithZLIB() appcontext.AppFunc
 // It will return a byte[] of the encrypted data.
 // This function is a configuration function and returns a function pointer.
 func (dynamic AppFunctionsSDKConfigurable) EncryptWithAES(parameters map[string]string) appcontext.AppFunction {
-	key, ok := parameters[Key]
-	if !ok {
-		dynamic.Sdk.LoggingClient.Error("Could not find " + Key)
+	// SecretPath & SecretName are optional if EncryptionKey specified
+	secretPath := parameters[SecretPath]
+	secretName := parameters[SecretName]
+
+	// EncryptionKey is optional if SecretPath & SecretName are specified
+	encryptionKey := parameters[EncryptionKey]
+
+	if len(encryptionKey) == 0 && (len(secretPath) == 0 || len(secretName) == 0) {
+		dynamic.Sdk.LoggingClient.Errorf("Could not find '%s' or '%s' and '%s' in configuration", EncryptionKey, SecretPath, SecretName)
+		return nil
+	}
+
+	if (len(secretPath) != 0 && len(secretName) == 0) || (len(secretPath) == 0 && len(secretName) != 0) {
+		dynamic.Sdk.LoggingClient.Errorf("'%s' and '%s' both must be set in configuration", SecretPath, SecretName)
```

Comment:
this looks hard to understand what you are checking: are you verifying all 3 of them are needed in the configuration file?

Reply:
No, `encryptionKey` can be used washout others or `secretPath/secretName` without `encryptionKey` See comments on line that extracts them from parameters.

Reply:
ok, you might want to put your comments just above the checking logic.